gnovm/pkg/gnolang/transcribe.go

```diff
@@ -210,6 +210,7 @@ func transcribe(t Transform, ns []Node, ftype TransField, index int, n Node, nc
 		}
 	case *StarExpr:
 		cnn.X = transcribe(t, nns, TRANS_STAR_X, 0, cnn.X, &c).(Expr)
+		cnn.IsLHS = ftype == TRANS_ASSIGN_LHS
```

Comment:
Again, it works well, but not sure this is the right place to do it. What about adding a TRAN_STAR_X section in the transform function in Process.go instead?

Reply:
will this be a better choice?

// TRANS_LEAVE -----------------------
			case *StarExpr:
				n.IsLHS = ftype == TRANS_ASSIGN_LHS

Reply:
I don't think so. I don't think `ftype` would have the value necessary if we put this in TRANS_LEAVE

Reply:
I think what @ltzmaxwell suggested is the right place to add in the TRANS_LEAVE section of transform function in preprocess.go 

here is the recursive tree to help us visualize it. 

![image](https://github.com/gnolang/gno/assets/90544084/0ca89efc-2eb0-465b-86d1-cb4856798a26)

Reply:
After further review, we can remove the `IsLHS bool` field in the `type StarExpr struct`,  the associated logic in preprocess here, and the checking in `PushForPointer()`, and `PopAsPointer()` entirely.

This is because PushForPointer and PopAsPointer should be, and are only used for, the left-hand side (LHS) of an assignment to update the LHS value. 

The rest of the logic are necessary and right on the spot. 

When StarExpr *el appears on the right-hand side (RHS) or stands alone, we push StarExpr and call doOpEval() to evaluate the StarExpr. In other words, we dereference el to get the underlying value.

When StarExpr *el appears on the LHS, we wrap it with RefExpr and OpRef to get the pointer value of the entire StarExpr *el and push it to the m.Value stack before it is updated with RHS values. In other words, we get the reference that the RHS points to, which is a pointer value, to *el.

Reply:
After further review, I think we can remove the added `IsLHS` bool field in the `type StarExpr struct`, the associated logic in preprocess here, and the checking in `PushForPointer()`, and `PopAsPointer()` entirely. This is because `PushForPointer` and `PopAsPointer` should be, and are only used for, the left-hand side (LHS) of an assignment to update the LHS value.

The rest of the logic is necessary and right on the spot.

When `StarExpr *el` appears on the LHS, we wrap it with `RefExpr` and push OpCode `OpRef` on the stack and get the pointer value of the `StarExpr *el` and then push it to the `m.Value` stack before it is updated with RHS values. In other words, we get the reference that the LHS points to, which is a pointer value, to the `StarExpr *el`.

There are other places considered as LHS in an assignment, such as increment and ranging through iterators.

When `StarExpr *el` appears on the right-hand side (RHS) or stands alone, we push `StarExpr` on the stack and call `doOpEval()` to evaluate the `StarExpr *el`. That is how we dereference `el` to get the underlying value, in the process.

Reply:
Thanks Ray. Done